### DISCUSSION TITLE
New argument: no_speech_strict_lvl

### DISCUSSION THREAD

**Discussion by Purfview** [Purfview/whisper-standalone-win#44] [Announcements]
`no_speech_strict_lvl` meant to combat cases where the model is getting stuck in a failure loop or outputs nonsense.
Use `whisper-faster.exe --help` to read more about it.
When its action kicks in you'll see debug log containing "strict" word [use `--verbose=true`].

Implemented since r141.1 and disabled by default. Use with `--beam_size=5` if  enabled.

**Comment by hoonlight** [comment]
Thank you for adding this useful feature. This seems like a good approach!  The default of 0.9 looks good.

**Reply by Purfview** [reply]
Did you encountered cases where it kicks in?

In `r141.2` I hard set it to 0.93, because all cases I encountered were above 0.94.

**Reply by hoonlight** [reply]
There were some false positives, but it generally worked well on my dataset. I'm sure you've tested it enough and updated it with more suitable values. I'll try to test the new values again as well. thanks

**Comment by despairTK** [comment]
The default settings for this new parameter may affect the normal generation of subtitles. I am currently using the latest version you released, and I basically generate subtitles with the following improved parameters:
--language=[According to the video language setting]
--model=[According to the video language setting]
--output_format srt
--compute_type float32
--initial_prompt "[Fill in a suitable example sentence according to the video language]"
--batch_recursive True

The new parameter "no_speech_strict_lvl" was not added when subtitles were generated last time, the probability of generating wrong subtitles is almost zero, and the speaker cannot be identified.

In the current latest version, if the parameter "no_speech_strict_lvl" is set as the default parameter, it is easy to cause no subtitles to be generated. Please see the screenshot below for details:
![QQ截图20230730231153](https://github.com/Purfview/whisper-standalone-win/assets/111871110/39eb2801-f406-44a3-8e97-363e85a23a41)
There is a large section of ungenerated subtitles in the picture, and some videos have more ungenerated parts than the screenshots. If the "no_speech_strict_lvl" parameter is changed to 0, everything works fine.

**Reply by Purfview** [reply]
Can you share audio sample?

Btw,  "--output_format srt" is redundant because this is default setting.

**Reply by despairTK** [reply]
Yes, I found one of the obvious and serious audio examples.
Language is English. I used the model of medium.en and Large-V2. In fact, the parameters are the same as previously provided.
[test.zip](https://github.com/Purfview/whisper-standalone-win/files/12211165/test.zip)

**Reply by Purfview** [reply]
Problem goes away if you set beam_size=5. Anyway, I disabled it by default in r141.4.

**Reply by raphaelomoreira** [reply]
Hello!
I'm having a similar problem.
I'm using the default parameters. I put --beam_size 5.
But I still have parts of the audio without recognition. This happens more in parts where there is a change of voice, when the speaker changes.

device=cuda
beam_size=5
no_speech_strict_lvl=0

I've already tried some small parameter changes, but nothing worked.
It always fails in the same place!
